### PR TITLE
Handle error responses from INH backend

### PR DESCRIPTION
**Pull request by rik86future**
Part of: https://trello.com/c/nQwfSTEP/69-fix-bug-form-update-on-inh-admin

Fixes, misleading Success screen when the error responses were not successful,
Fixes a missing "uprn" parameter which led to the backend creating records as "Exceptions"

image

image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@ axeReports/
 *.zip
 sassdoc/
 .env
+.idea
```
Comment (kjdchapman) [inline .gitignore:10]: nit: needs newline at end of file

File: controllers/help-requests/help_requests.controller.js [hunk 2]
```diff
@@ -175,6 +175,14 @@ module.exports = {
 
                 await HelpRequestsService.createHelpRequest(query, userName)
                 .then(result => {
+                    if(result.isError === true){
+                        let errorMessage = result.data || "Server error, please try later."
```
Comment (kjdchapman) [inline controllers/help-requests/help_requests.controller.js:179]: I wouldn't show users of this service the raw error message - it'll probably be something technical and scary.

I suggest:
```suggestion
                        let errorMessage = "There's a problem with our service. Call our helpline at xxx-xxx-xxxx."
```

The tradeoff is we "lose" what these errors were, and they're harder to debug, but I think it's more important that a user (especially one in need of help!) doesn't get stuck.

File: controllers/help-requests/help_requests.controller.js [hunk 3]
```diff
@@ -202,12 +210,11 @@ module.exports = {
 
         if (!errors.isEmpty()) {
             var extractedErrors = mapFieldErrors(errors);
-
+            let errorMessage = "Please complete all fields marked in red before updating.";
```
Comment (kjdchapman) [inline controllers/help-requests/help_requests.controller.js:213]: nit: I'd use language as suggested in [the GOV.UK guidance](https://design-system.service.gov.uk/components/error-message/). 

What we have here is closer to the [error summary component](https://design-system.service.gov.uk/components/error-summary/), as we don't have the tech for individual error messages yet.
```suggestion
            let errorMessage = "Fill in the fields marked in red.";
```

File: controllers/help-requests/help_requests.controller.js [hunk 4]
```diff
@@ -217,7 +224,12 @@ module.exports = {
 
                 await HelpRequestsService.updateHelpRequest(query, userName)
                 .then(result => {
-                    return res.redirect('/help-requests/edit/' + query.Id + "?hasupdated=true");
+                    if(result.isError === true){
+                        let errorMessage = result.data || "Server error, please try later."
```
Comment (kjdchapman) [inline controllers/help-requests/help_requests.controller.js:228]: Same suggestion as above.

File: public/app.css [hunk 5]
```diff
@@ -1,4 +1,4 @@
-@import url("https://fonts.googleapis.com/css?family=Montserrat:300,400,500,700&display=swap");
+@import url("https://fonts.googleapis.com/css2?family=Open+Sans:300,400,600,700&display=swap");
```
Comment (kjdchapman) [inline public/app.css:1]: It'd be good to do this as a separate commit/PR in the future (I don't think it's worth the effort of splitting it out now though).

File: controllers/help-requests/help_requests.controller.js (hunk 3 above)
Comment (kjdchapman) [reply controllers/help-requests/help_requests.controller.js:213]: Ah, I see this was already the error message before these changes.

I think it's still worth the effort to change whilst we're here, but up to you.

File: controllers/help-requests/help_requests.controller.js (hunk 2 above)
Comment (rik86future) [reply controllers/help-requests/help_requests.controller.js:179]: I don't know a phone number to put in there, maybe "Sorry, there is a problem with the service. Try again later" https://design-system.service.gov.uk/patterns/problem-with-the-service-pages/

File: controllers/help-requests/help_requests.controller.js (hunk 3 above)
Comment (rik86future) [reply controllers/help-requests/help_requests.controller.js:213]: no problem, I'll have a go now

File: public/app.css (hunk 5 above)
Comment (rik86future) [reply public/app.css:1]: oh no, this is my bad it's from another pr

Comment (rik86future) [reply public/app.css:1]: I'll revert

File: controllers/help-requests/help_requests.controller.js (hunk 2 above)
Comment (kjdchapman) [reply controllers/help-requests/help_requests.controller.js]: We could ask the team about this? 

Alternately, I think your suggestion is a good one - wouldn't argue with just using that for now, to get this done.

**Pull request merged**